packages/auth/test/integration/flows/totp.test.ts

```diff
@@ -0,0 +1,147 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+ import { expect, use } from 'chai';
+ import chaiAsPromised from 'chai-as-promised';
+ import * as sinon from 'sinon';
+ import sinonChai from 'sinon-chai';
+ 
+//import { mockTotp } from '../../helpers/integration/helpers';
+import {Auth, createUserWithEmailAndPassword, multiFactor, signInAnonymously, signInWithEmailAndPassword, UserCredential} from '@firebase/auth';
+
+import {
+  cleanUpTestInstance,
+  getTestInstance,
+  mockTotp,
+  randomEmail
+} from '../../helpers/integration/helpers';
+import { MultiFactorAssertionImpl } from '../../../src/mfa/mfa_assertion';
+
+import { MultiFactorSessionImpl } from '../../../src/mfa/mfa_session';
+import { TotpMultiFactorAssertionImpl, TotpMultiFactorGenerator, TotpSecret } from '../../../src/mfa/assertions/totp';
+import * as MFA from '../../../src/api/account_management/mfa';
+import { FirebaseError } from '@firebase/util';
+
+
+
+use(chaiAsPromised);
+use(sinonChai);
+
+const TOTP_COMB_A = {
+
+    response: { sharedSecretKey: 'secretKey3',
+              verificationCodeLength: 30,
+              hashingAlgorithm: 'sha1',
+              periodSec:30,
+              sessionInfo: 'testsSessionInfo',
+              finalizeEnrollmentTime:  Date.now()
+              },
+    code: '...'
+  };
+  
+  const TOTP_COMB_B = {
+
+    response: { sharedSecretKey: 'secretKey2',
+    verificationCodeLength: 30,
+    hashingAlgorithm: 'sha1',
+    periodSec: 30,
+    sessionInfo: 'testsSessionInfo',
+    finalizeEnrollmentTime:  Date.now()
+    },
+    code: '...'
+  };
+
+describe(' Integration tests: Mfa TOTP', () => {
```

Comment:
I suggest that we have the following test cases inside this:

1) Create user with email/password + enroll in TOTP + login with correct code - success case end-to-end
   In this test, we make sure that enrollment succeeds, then complete login and check that the error 'auth/multi-factor-auth-required' is thrown and supply the otp code - similar to this sample - https://firebase.google.com/docs/auth/web/multi-factor#signing_users_in_with_a_second_factor, but with totp.
2) Create user with email/password + enroll in TOTP +login with incorrect code - fails login. We can reuse the same email/password in test case 1 for this.

3) Unenroll from TOTP for the test case 1 email and ensure that sign in succeeds without asking for OTP code i.e no 'auth/multi-factor-auth-required' error.

You can refer to this test - https://github.com/firebase/firebase-js-sdk/blob/e9fb4037f3908d2a05070de9219aa8a94d287f14/packages/auth/test/integration/flows/email.test.ts#L85, which reuses the same email/password user to test more than one case.


4) Create user with email/password + enroll in TOTP with wrong OTP code - enrollment fails.